include/filewriter/filewriter_impl.hpp

```diff
@@ -65,32 +70,62 @@ void FileWriter::WriteSummary()
   }
 }
 
-void FileWriter::CreateEvent(size_t step, mlboard::Summary *summary)
+inline void FileWriter::CreateEvent(size_t step, mlboard::Summary *summary)
 {
-    Event event;
-    double wall_time = time(nullptr);
-    event.set_wall_time(wall_time);
-    event.set_step(step);
-    event.set_allocated_summary(summary);
-    q.Push(event);
+  Event event;
+  double wall_time = time(nullptr);
+  event.set_wall_time(wall_time);
+  event.set_step(step);
+  event.set_allocated_summary(summary);
+  q.Push(event);
 }
 
-void FileWriter::Flush()
+inline void FileWriter::Flush()
 {
   // Flush everything successfully and close the thread.
   thread_->join();
 }
 
-void FileWriter::Close()
+inline void FileWriter::Close()
 {
   close_ = false;
   Flush();
 }
 
-FileWriter::~FileWriter()
+inline FileWriter::~FileWriter()
 {
   if (close_)
     Close();
+
+  std::regex regExp("(_preprocess)(.*)");
+
+  // Delete all the directories created during processing.
+  struct dirent *entry;
+  DIR *dir = opendir("./");
+
+  if (dir == NULL)
+  {
+    return;
+  }
+  while ((entry = readdir(dir)) != NULL)
+  {
+    if (entry->d_name != "." && entry->d_name != ".."
```

Comment:
Hey I think it's better we remove _preprocess directory after every image logging is done. It would help in less diskusage of user. Let me know what you think.

Reply:
Umm, the directory would be empty, so i think it is ok, I think so if it is outside the we can use it for other purpose too , and be sure that all the files are deleted and won't face the issue which we were facing when you ported this function outside the test to main file

Reply:
This doesn't remove the directory in my os. But let's keep it for now.